invoice/src/main/java/org/killbill/billing/invoice/usage/ContiguousIntervalUsageInArrear.java

```diff
@@ -376,9 +402,12 @@ BigDecimal computeToBeBilledCapacityInArrear(final List<RolledUpUnit> roUnits) t
                     complies = false;
                     break;
                 }
+                toBeBilledDetails.add(new ConsumableInArrearDetail(tierNum++, ro.getUnitType(), cur.getRecurringPrice().getPrice(getCurrency()), ro.getAmount().intValue(), BigDecimal.ZERO, null, null));
```

Comment:
I am wondering if this actually should be outside of that loop, in its own separate loop: if there are three different units, the first two comply but not the third one, we would add these details nonetheless (instead of adding them only for the correct tier).

It would be good to verify this behavior in `TestContiguousIntervalCapacityInArrear`.

Reply:
Using the `TestContiguousIntervalCapacityInArrear`,  the details that is been added (first result) is 

> [{"tier":1, "tierUnit":"unit1", "tierPrice":10, "quantity":100, "reference":null, "existingUsageAmount":null, "amount":0},{"tier":2, "tierUnit":"unit2", "tierPrice":10, "quantity":1000, "reference":null, "existingUsageAmount":null, "amount":10}]

I modified the test to add other unit and this is the detail that produces:
> [{"tier":1, "tierUnit":"unit1", "tierPrice":10, "quantity":100, "reference":null, "existingUsageAmount":null, "amount":0},{"tier":2, "tierUnit":"unit2", "tierPrice":10, "quantity":1000, "reference":null, "existingUsageAmount":null, "amount":0},
{"tier":3, "tierUnit":"unit3", "tierPrice":10, "quantity":50, "reference":null, "existingUsageAmount":null, "amount":10}]